File: FrontEnd_Shipping/my-app/src/types/shipping/ShipmentResponse.ts
import type { ContactInfo } from "./ContactInfo";

export interface ShipmentResponse {
  shipmentId: string;
  userId: number;

  status: string;

  sender: ContactInfo;
  receiver: ContactInfo;

  trackingNumber: string;
  carrier: string;

  weight: number;

  statusHistory: Array<Record<string, unknown>>;
  files: Record<string, Record<string, unknown>>;

  createdAt: string;   // Instant → ISO string
  updatedAt: string;
}

File: basic-java-main/shipping-micro-service/src/main/java/com/example/shippingService/dtos/ShipmentResponse.java
package com.example.shippingService.dtos;

import com.fasterxml.jackson.annotation.JsonAlias;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.time.Instant;
import java.util.List;
import java.util.Map;


@Data
public class ShipmentResponse {

    private String shipmentId;
    private Long userId;

    private String status;

    private ContactInfo sender;
    private ContactInfo receiver;

    private String trackingNumber;
    private String carrier;

    @JsonAlias({"weight_kg", "weight"})
    private Double weight;

    private List<Map<String, Object>> statusHistory;
    private Map<String, Map<String, Object>> files;

    private Instant createdAt;
    private Instant updatedAt;

    public ShipmentResponse() {}
}
